tests/lib/rules/func-call-spacing.js

```diff
@@ -0,0 +1,378 @@
+/**
+ * @fileoverview Tests for func-call-spacing rule.
+ * @author Matt DuVall <http://www.mattduvall.com>
+ */
+
+"use strict";
+
+//------------------------------------------------------------------------------
+// Requirements
+//------------------------------------------------------------------------------
+
+var rule = require("../../../lib/rules/func-call-spacing"),
+    RuleTester = require("../../../lib/testers/rule-tester");
+
+//------------------------------------------------------------------------------
+// Tests
+//------------------------------------------------------------------------------
+
+var ruleTester = new RuleTester();
+
+ruleTester.run("func-call-spacing", rule, {
+    valid: [
+
+        // default ("never")
+        "f();",
+        "f(a, b);",
+        "f.b();",
+        "f.b().c();",
+        "f()()",
+        "(function() {}())",
+        "var f = new Foo()",
+        "var f = new Foo",
+        "f( (0) )",
+        "( f )( 0 )",
+        "( (f) )( (0) )",
+        "( f()() )(0)",
+        "(function(){ if (foo) { bar(); } }());",
+        "f(0, (1))",
+        "describe/**/('foo', function () {});",
+        "new (foo())",
+
+        // "never"
+        {
+            code: "f();",
+            options: ["never"]
+        },
+        {
+            code: "f(a, b);",
+            options: ["never"]
+        },
+        {
+            code: "f.b();",
+            options: ["never"]
+        },
+        {
+            code: "f.b().c();",
+            options: ["never"]
+        },
+        {
+            code: "f()()",
+            options: ["never"]
+        },
+        {
+            code: "(function() {}())",
+            options: ["never"]
+        },
+        {
+            code: "var f = new Foo()",
+            options: ["never"]
+        },
+        {
+            code: "var f = new Foo",
+            options: ["never"]
+        },
+        {
+            code: "f( (0) )",
+            options: ["never"]
+        },
+        {
+            code: "( f )( 0 )",
+            options: ["never"]
+        },
+        {
+            code: "( (f) )( (0) )",
+            options: ["never"]
+        },
+        {
+            code: "( f()() )(0)",
+            options: ["never"]
+        },
+        {
+            code: "(function(){ if (foo) { bar(); } }());",
+            options: ["never"]
+        },
+        {
+            code: "f(0, (1))",
+            options: ["never"]
+        },
+        {
+            code: "describe/**/('foo', function () {});",
+            options: ["never"]
+        },
+        {
+            code: "new (foo())",
+            options: ["never"]
+        },
+
+        // "always"
+        {
+            code: "f ();",
+            options: ["always"]
+        },
+        {
+            code: "f (a, b);",
+            options: ["always"]
+        },
+        {
+            code: "f\n();",
```

Comment:
Yes, but a syntactic declaration (like an arrow or generator) *can* span multiple lines, but if you do `f\n(g)` I think there might be ASI issues.